apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/charSequenceContainsCreators.kt

```diff
@@ -190,6 +190,32 @@ fun <T : CharSequence> CharSequenceContains.CheckerOption<T, NoOpSearchBehaviour
     vararg otherPatterns: String
 ): Expect<T> = addAssertion(ExpectImpl.charSequence.contains.regex(this, pattern glue otherPatterns))
 
+/**
+ * Finishes the specification of the sophisticated `contains` assertion where the given regular expression [pattern]
+ * as well as the [otherPatterns] are expected to have a match, using a non disjoint search.
+ *
+ * By non disjoint is meant that `'aa'` in `'aaaa'` is found three times and not only two times.
+ * Also notice, that it does not search for unique matches. Meaning, if the input of the search is `'ab'` and [pattern]
+ * is defined as `'a(b)?'` and one of the [otherPatterns] is defined as `'a(b)?'` as well, then both match, even though
+ * they match the same sequence in the input of the search. Use an option such as [atLeast], [atMost] and [exactly] to
+ * control the number of occurrences you expect.
+ *
+ * Meaning you might want to use:
+ *   `contains.exactly(2).regex('a(b)?')`
+ * instead of:
+ *   `contains.atLeast(1).regex('a(b)?', 'a(b)?')`
+ *
+ * @param pattern The pattern which is expected to have a match against the input of the search.
+ * @param otherPatterns Additional patterns which are expected to have a match against the input of the search.
+ *
+ * @return The [Expect] for which the assertion was built to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ */
+fun <T : CharSequence> CharSequenceContains.CheckerOption<T, NoOpSearchBehaviour>.regex(
+    pattern: Regex,
+    vararg otherPatterns: Regex
+): Expect<T> = regex(pattern.toString(), *otherPatterns.map { it.toString() }.toTypedArray())
```

Comment:
You have implemented an overload for ExpectImpl.charSequence.contains.regex which expects `Regex` please use it here